addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -99,23 +111,64 @@ public static boolean contains(String tag) {
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
+        SiteMap tree = Model.getSingleton().getSession().getSiteTree();
         List<Element> sourceElements = source.getAllElements();
         sourceElements.stream()
                 .filter(element -> SupportedElements.contains(element.getName()))
                 .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
                 .forEach(
-                        element ->
-                                newAlert()
-                                        .setRisk(Alert.RISK_MEDIUM)
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            newAlert()
+                                    .setRisk(Alert.RISK_MEDIUM)
+                                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                                    .setDescription(getString("desc"))
+                                    .setSolution(getString("soln"))
+                                    .setReference(getString("refs"))
+                                    .setEvidence(element.toString())
+                                    .setCweId(345) // CWE-345: Insufficient Verification of Data
+                                    // Authenticity
+                                    .setWascId(15) // Application Misconfiguration
+                                    .setOtherInfo(
+                                            Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherInfo",
+                                                    calculateIntegrityHash(msg, element, tree)))
+                                    .raise();
+                        });
```

Comment:
This is still essentially a single statement, the curly braces shouldn't be needed unless I'm missing something.

Reply:
Actually the braces probably are necessary in order to check if the hash was calculated successfully.

```
String hash = calculateIntegrityHash(msg, element, tree);
String otherInfo = hash.isEmpty ? "" : Constant.messages.getString(MESSAGE_PREFIX + "otherInfo", hash);
...
.setOtherInfo(otherInfo);
...
```

(That assumes it's set empty `""` initially, if you use null then `== null`...)

Reply:
Alright I actually thought of doing it. But since I'm new I wanted your opinion on it. I'll add the changes.